nmosnode/api.py

```diff
@@ -54,19 +55,29 @@ def versionroot(self, api_version):
             abort(404)
         return ["self/","sources/", "flows/", "devices/", "senders/", "receivers/"]
 
-    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/senders/")
-    def senders(self, api_version):
+    @route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/self/")
+    def selfresource(self, api_version):
         if api_version not in NODE_APIVERSIONS:
             abort(404)
-        return self.registry.list_resource("sender", api_version=api_version).values()
+        return self.registry.list_self(api_version=api_version)
+
+    @resource_route('/'+NODE_APINAMESPACE+'/'+NODE_APINAME+"/<api_version>/<resource_type>/")
```

Comment:
Flask match priority seems to be complicated, but I'm kinda worried this route also matches the /self/ route defined above - meaning that there is the potential for flask to decide to use this method for /self and end up with this kind of issue: https://stackoverflow.com/questions/17868916/flask-route-pattern-matching-order

Reply:
I wonder if they've improved the logic since then. I could easily do this via a single method if required.